docs: Update Vercel deployment documentation. (#3000)
nuxt/framework: docs/content/3.docs/3.deployment/6.vercel.md

# Vercel

> Nuxt on Vercel supports both static and server-rendered pages.

Reviewer: At the moment Nuxt 3 / Nitro does is not implementing static generation. Also we support API routes maybe reword like this:

```suggestion
Nuxt on Vercel supports server-rendered pages and API routes.
```

- No configuration required
::

## CLI

1. Install the [Vercel CLI](https://vercel.com/cli).
2. Vercel will detect that you are using Nuxt and will enable the correct settings for your deployment.

> 3. Your application is deployed! (e.g. [nuxt.vercel.app](https://nuxt.vercel.app/))

Reviewer: Can we please have access to this deployment? (ideally on vercel.com/nuxt-js)

Author: It's coming from here: https://github.com/vercel/vercel/tree/main/examples/nuxtjs

Which is our "single source of truth" for /templates

```bash
$ npm i -g vercel

> $ vercel init nuxt

Reviewer: I prefer if we use `nuxi` cli with [vercel starter](https://github.com/nuxt/starter/tree/v3-vercel): `npx nuxi init -t v3-vercel`

Author: `vercel init nuxt` creates a project that uses that same code, from here https://github.com/vercel/vercel/tree/main/examples/nuxtjs

Thoughts?

Reviewer: Having an example deployment is nice in vertical but `nuxi init` should be used to make a new nuxt project. Since vercel integration is zero-config, a project made with `nuxi init` should be deployable to vercel via cli as well.

<img width="905" alt="image" src="https://user-images.githubusercontent.com/5158436/151834208-94e0281a-10b5-482f-ba1b-7e9c79f27f7f.png">

> $ vercel init nuxtjs

Reviewer: ```suggestion
$ vercel init nuxt
```

Vercel CLI

> > Success! Initialized "nuxtjs" example in ~/your-folder.

Reviewer: ```suggestion
> Success! Initialized "nuxt" example in ~/your-folder.
```

> ```bash
> $ npm i -g vercel
> $ vercel init nuxt
> Vercel CLI
> > Success! Initialized "nuxt" example in ~/your-folder.
> - To deploy, `cd nuxt` and run `vercel`.
> ```

Author: ```suggestion
```bash
$ npm i -g vercel
$ npx nuxi init -t v3-vercel
$ vercel
```
```

> - To deploy, `cd nuxtjs` and run `vercel`.

Reviewer: ```suggestion
- To deploy, `cd nuxt` and run `vercel`.
```

Author: It's actually https://github.com/vercel/vercel/tree/main/examples/nuxtjs here
